core/src/main/java/org/jboss/gwt/elemento/core/BodyObserver.java

```diff
@@ -0,0 +1,149 @@
+package org.jboss.gwt.elemento.core;
+
+import elemental2.dom.*;
+import jsinterop.base.Js;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static elemental2.dom.DomGlobal.document;
+import static java.util.Objects.isNull;
+import static java.util.Objects.nonNull;
+
+class BodyObserver {
+
+    private static String ATTACH_UID_KEY = "on-attach-uid";
+    private static String DETACH_UID_KEY = "on-detach-uid";
+
+    private static List<ElementObserver> detachObservers = new ArrayList<>();
+    private static List<ElementObserver> attachObservers = new ArrayList<>();
+
+    static {
+        MutationObserver mutationObserver = new MutationObserver((MutationRecord[] records, MutationObserver observer) -> {
+            for (int i = 0; i < records.length; i++) {
+                onElementsRemoved(records[i]);
+                onElementsAppended(records[i]);
+            }
```

Comment:
Isn't it better to use foreach loops?

```
for (MutationRecord record : records) {
    onElementsRemoved(record);
    onElementsAppended(record);
}
```

Reply:
Normally normal loops should be faster, but i think that is in the case of streams not the for-each loops so i will check and update the code accordingly.

Reply:
i don't care if you use normal loops or foreach. i prefer foreach over "normal loops" because of the less mistakes that can be made, but i don't know what is faster. whatever you prefer ;)